rtrlib/rtr/packets.c

```diff
@@ -923,10 +977,98 @@ int rtr_sync_receive_and_store_pdus(struct rtr_socket *rtr_socket){
+                        // if interval values are out of bounds, ignore them without error PDU.
+                        break;
+                    default:
+                        // if interval values are out of bound and no option is set, send error PDU
+                        RTR_DBG("Received expiration value out of range. Was %u, must be between %u and %u.",
+                                ((struct pdu_end_of_data_v1 *) pdu)->expire_interval, EXPIRATION_MIN, EXPIRATION_MAX);
+                        const char txt[] = "Interval value out of range";
+                        rtr_send_error_pdu(rtr_socket, pdu, RTR_MAX_PDU_LEN, CORRUPT_DATA, txt, strlen(txt) + 1);
+                        retval = RTR_ERROR;
+                        goto cleanup;
+                        break;
+                    }
+                }
+
+                interv_retval = rtr_check_interval_range(((struct pdu_end_of_data_v1 *) pdu)->refresh_interval, REFRESH);
+                if (interv_retval == INSIDE_INTERVAL_RANGE) {
+                    rtr_socket->refresh_interval = ((struct pdu_end_of_data_v1 *) pdu)->refresh_interval;
+                } else {
+                    switch (INTERVAL_OPTION) {
+                    case DEFAULT_MIN_MAX:
+                        if (interv_retval == BELOW_INTERVAL_RANGE) {
+                            rtr_socket->refresh_interval = REFRESH_MIN;
+                        } else {
+                            rtr_socket->refresh_interval = REFRESH_MAX;
+                        }
+                        break;
+                    case ACCEPT_ANY:
+                        rtr_socket->refresh_interval = ((struct pdu_end_of_data_v1 *) pdu)->refresh_interval;
+                        break;
+                    case IGNORE_ON_FAILURE:
+                        // if interval values are out of bounds, ignore them without error PDU.
+                        break;
+                    default:
+                        // if interval values are out of bound and no option is set, send error PDU
+                        RTR_DBG("Received refresh value out of range. Was %u, must be between %u and %u.",
+                                ((struct pdu_end_of_data_v1 *) pdu)->refresh_interval, REFRESH_MIN, REFRESH_MAX);
+                        const char txt[] = "Interval value out of range";
+                        rtr_send_error_pdu(rtr_socket, pdu, RTR_MAX_PDU_LEN, CORRUPT_DATA, txt, strlen(txt) + 1);
+                        retval = RTR_ERROR;
+                        goto cleanup;
+                        break;
+                    }
+                }
+
+                interv_retval = rtr_check_interval_range(((struct pdu_end_of_data_v1 *) pdu)->refresh_interval, RETRY);
+                if (interv_retval == INSIDE_INTERVAL_RANGE) {
+                    rtr_socket->retry_interval = ((struct pdu_end_of_data_v1 *) pdu)->retry_interval;
+                } else {
+                    switch (INTERVAL_OPTION) {
```

Comment:
Here is a lot of code duplication to handle the different intervals according to the interval_option.

Either write another helper function that just returns the correct value corresponding to `interval_option` and `interval_type` or expand/rewrite `rtr_check_interval_range` to return the value instead of just `-1`, `0`, or `1`,